src/display.jl

```diff
@@ -1,379 +1,405 @@
-        elseif format == :midpoint
-            return string("(", representation(x, format), ")", n)
         else
-            return string(representation(x, format), n)
+            new_mantissa = string(
+                    view(mantissa, 1:idx-1),
+                    parse(Int, mantissa[idx]) - 1,
+                    # Add '.' if the last significant digit not equal to 0 is before the decimal point
+                    idx == 1 ? "." : "",
+                    '9'^(len - (idx + 1)))
+            return string(new_mantissa, 'e', exponent)
+
         end
     end
+end
 
-    if format == :full
-        full_str = representation.(X.v, :full)
-        return string("IntervalBox(", join(full_str, ", "), ")")
-    elseif format == :midpoint
-        return string("(", join(X.v, ") × ("), ")")
+# Utilities
+
+function subscriptify(n::Integer)
+    if 0 ≤ n ≤ 9
+        return subscript_digit(n)
     else
-        return join(X.v, " × ")
+        len = ndigits(n)
+        x = Vector{String}(undef, len)
+        i = 0
+        while n > 0
+            n, d = divrem(n, 10)
+            x[len-i] = subscript_digit(d)
+            i += 1
+        end
+        return join(x)
     end
-
 end
 
-function representation(x::Complex{<:Interval}, format=nothing)
-    if format === nothing
-        format = display_params.format
+function supscriptify(n::Integer)
+    if 0 ≤ n ≤ 9
+        return supscript_digit(n)
+    else
+        len = ndigits(n)
+        x = Vector{String}(undef, len)
+        i = 0
+        while n > 0
+            n, d = divrem(n, 10)
+            x[len-i] = supscript_digit(d)
+            i += 1
+        end
+        return join(x)
     end
-
-    format == :midpoint && return string('(', x.re, ')', " + ", '(', x.im, ')', "im")
-
-    return string(x.re, " + ", x.im, "im")
 end
 
-for T in (Interval, DecoratedInterval)
-    @eval show(io::IO, a::$T{S}) where S = print(io, representation(a))
-    @eval showfull(io::IO, a::$T{S}) where S = print(io, representation(a, :full))
-    @eval showfull(a::$T{S}) where S = showfull(stdout, a)
+function supscript_digit(i::Integer)
+    i == 0 && return "⁰"
+    i == 1 && return "¹"
+    i == 2 && return "²"
+    i == 3 && return "³"
+    i == 4 && return "⁴"
+    i == 5 && return "⁵"
+    i == 6 && return "⁶"
+    i == 7 && return "⁷"
+    i == 8 && return "⁸"
+    return "⁹"
 end
 
-for T in (IntervalBox, Complex{<:Interval})
-    @eval show(io::IO, a::$T) = print(io, representation(a))
-    @eval show(io::IO, ::MIME"text/plain", a::$T) = print(io, representation(a))
-    @eval showfull(io::IO, a::$T) = print(io, representation(a, :full))
-    @eval showfull(a::$T) = showfull(stdout, a)
+function subscript_digit(i::Integer)
+    i == 0 && return "₀"
+    i == 1 && return "₁"
+    i == 2 && return "₂"
+    i == 3 && return "₃"
+    i == 4 && return "₄"
+    i == 5 && return "₅"
+    i == 6 && return "₆"
+    i == 7 && return "₇"
+    i == 8 && return "₈"
+    return "₉"
```

Comment:
I would also include the

```julia
i == 9 && return "₉"
```

and throw an exception as fallback. Might be a little overcautious, but if somehow you get an invalid integer and the printing silently,identifying the bug might be a little tricky

Reply:
That's a good point. Done.